test/test_other.py

```diff
@@ -2175,6 +2175,29 @@ def test(link_flags, lib_suffix):
     test(['-lfile'], '') # -l, auto detection from library path
     test(['libdir/libfile.so.3.1.4.1.5.9'], '.3.1.4.1.5.9') # handle libX.so.1.2.3 as well
 
+  @node_pthreads
+  @also_with_modularize
+  def test_dylink_pthread(self):
+    create_file('library.c', r'''
+      int answer() {
+        return 42;
+      }
+    ''')
+    self.run_process([EMCC, 'library.c', '-pthread', '-Wno-experimental', '-sSIDE_MODULE', '-o', 'side.wasm'])
+    create_file('main.c', r'''
+      #include <assert.h>
+      #include <stdio.h>
+
+      extern int answer();
+
+      int main() {
+        assert(answer() == 42);
+        printf("done\n");
+        return 0;
+      }
+    ''')
+    self.do_runf('main.c', 'done\n', emcc_args=['-sMAIN_MODULE=2', '-pthread', '-Wno-experimental', 'side.wasm'])
```

Comment:
Do we need to write a whole new test here?  Is there not an existing dylink pthread test we could just add `@also_with_modularize` to?

Reply:
Ah, we could just merge it with the test below. Done with eea195e12fbe70d7ff5c06af6574e695d5d8f0d3.